### PR TITLE
Backend work

### PR DESCRIPTION
**Pull request by pavan4**
Backend for accepting submissions from MMIW Mobile apps:
* docker-compose based deploy 
* Works with Digital Ocean spaces 
* Rate limited API for submission 
* Database persists on the host volume 
* Auto cert based https deploy with LetsEncrypt
* Delete the older images if submission image is updated

TODO:
* Verify the submission

### CODE REVIEW

File: mmiwg2s-backend/controllers/submissionController.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        return res.status(500).json({
+          message: 'Error when deleting the submission.',
+          error: err
+        });
+      }
+
+      return res.status(204).json();
+    });
+  }
+  */
```
Comment (Mehequanna) [inline mmiwg2s-backend/controllers/submissionController.js:186]: Oh Nice! I was going to ask if there was a way to delete. I see it's commented out, that's ok. I'm glad to know it exists :)

File: mmiwg2s-backend/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    depends_on:
+      - "database"
+
+  database:
+    image: mongo:4.0.9
+    container_name: 'submission-database'
+    volumes:
+      - ./mongo-volume:/data/db
+    ports:
+      - 27017:27017
```
Comment (Mehequanna) [inline mmiwg2s-backend/docker-compose.yml:30]: Github says this needs a new line at the end.

File: mmiwg2s-backend/init-mongo.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    user: "test",
+    pwd: "secret",
+    roles : [
+      {
+        role: "readWrite",
+        db: "mmiwg2s-submission-database"
+      }
+    ]
+  }
+)
```
Comment (Mehequanna) [inline mmiwg2s-backend/init-mongo.js:12]: Github says this needs a new line at the end.

File: mmiwg2s-backend/init-mongo.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+db.createUser(
+  {
+    user: "test",
+    pwd: "secret",
```
Comment (Mehequanna) [inline mmiwg2s-backend/init-mongo.js:4]: Will these remain the same? Is this a security issue?

File: mmiwg2s-backend/routes/submissionRoutes.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+ * This route is unused now
+ * TODO update the verified or not using this
+ */
+// router.put('/:id', submissionController.update);
+
+/*
+ * DELETE
+ * We do not want to delete any entries now
+ */
+// router.delete('/:id', submissionController.remove);
```
Comment (Mehequanna) [inline mmiwg2s-backend/routes/submissionRoutes.js:46]: Makes sense. I'm curious, is there are way to make this harder to get to or do? This is not something to deal with now, just curious.

**Review by Mehequanna — APPROVED** [review]
This is great! Most of my comments are about new lines Github is suggesting :)

File: mmiwg2s-backend/init-mongo.js (hunk 4 above)
Comment (pavan4) [reply mmiwg2s-backend/init-mongo.js]: Not using this. I removed it.

File: mmiwg2s-backend/routes/submissionRoutes.js (hunk 5 above)
Comment (pavan4) [reply mmiwg2s-backend/routes/submissionRoutes.js:46]: It's not hard at all. Just have to enable the route with some access mechanism.

File: mmiwg2s-backend/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+Accept submission
+
+```
+POST
+/submissions
+params:
+name - Name [String] for submission
+email - email address [String] for submission
+image - file <max 5 MB only png|jpeg|jpg|webp>
```
[the remark this answers is not in the document]
Comment (pavan4) [reply mmiwg2s-backend/README.md:23]: Yes, I will make this configurable at the deploy level.

**Pull request merged**